containers-toolkit/Public/ContainerdTools.psm1

```diff
@@ -60,11 +61,48 @@ function Install-Containerd {
 
     process {
         if ($PSCmdlet.ShouldProcess($env:COMPUTERNAME, $WhatIfMessage)) {
+            $latestVersion = Get-ContainerdLatestVersion
+
+            # Get Containerd version to install
+            if (!$Version) {
+                # Get default version
+                $Version = $latestVersion
+            }
+            $Version = $Version.TrimStart('v')
+
+            # Check if a newer version is available
+            $userVersion = $Version
+            if ($Version -ne 'latest') {
+                Test-IsLatestVersion -Tool 'Containerd' -Version $Version -LatestVersion $latestVersion | Out-Null
+            }
+            else {
+                $userVersion = $latestVersion
+            }
+
             # Check if tool already exists at specified location
             if ($isInstalled) {
                 $errMsg = "Containerd already exists at $InstallPath or the directory is not empty"
                 Write-Warning $errMsg
 
+                # Check if user wants to install an already installed version
+                Write-Debug "Containerd executable: $ctrexe"
+                $cmdOutput = Invoke-ExecutableCommand -Executable "$ctrexe" -Arguments "--version"
```

Comment:
containerd and buildkit version checks look similar, perhaps break this logic out into utilities